core/src/main/java/org/wso2/testgrid/core/configchangeset/ConfigChangeSetExecutorUnix.java

```diff
@@ -0,0 +1,239 @@
+        }
+    }
+
+    /**
+     * Initialize agent before running config change set
+     *
+     * @param testPlan  The test plan
+     * @return          True if execution success. Else, false
+     */
+    @Override
+    public boolean initConfigChangeSet(TestPlan testPlan) {
+        try {
+            URL configChangeSetRepoPath = new URL(testPlan.getConfigChangeSetRepository());
+            String filePath = configChangeSetRepoPath.getPath();
+            String fileName = filePath.substring(filePath.lastIndexOf('/') + 1);
+            List<String> initShellCommand = new ArrayList<>();
+            initShellCommand.add("mkdir repos");
+            initShellCommand.add("cd repos && curl -LJO " +
+                    testPlan.getConfigChangeSetRepository() + "/archive/master.tar.gz   &>/dev/null");
+            initShellCommand.add("cd repos && tar xvzf " + fileName + "-master.tar.gz  &>/dev/null");
+            initShellCommand.add("chmod -R 755 repos/" + fileName + "-master/config-sets/ &>/dev/null");
+            return applyShellCommandOnAgent(testPlan, initShellCommand);
+        } catch (MalformedURLException e) {
+            logger.warn("Error parsing scenario repository path for ".
+                    concat(testPlan.getConfigChangeSetRepository()), e);
+            return false;
+        }
+    }
+
+    /**
+     * Revert back changes did in initConfigChangeSet
+     *
+     * @param testPlan  The test plan
+     * @return          True if execution success. Else, false
+     */
+    @Override
+    public boolean deInitConfigChangeSet(TestPlan testPlan) {
+        List<String> deInitShellCommand = new ArrayList<>();
+        deInitShellCommand.add("rm -rf repos");
+        return applyShellCommandOnAgent(testPlan, deInitShellCommand);
+    }
+
+
+    /**
+     * Send an array of shell command to an agent
+     *
+     * @param testPlan          The test plan
+     * @param initShellCommand  List of shell command
+     * @return  True if execution success. Else, false
+     */
+    private boolean applyShellCommandOnAgent(TestPlan testPlan, List<String> initShellCommand) {
+        String tinkererHost = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_REST_BASE_PATH);
+        // Get authentication detail from config.properties and encode with BASE64
+        String authenticationString = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_USERNAME) + ":" +
+                ConfigurationContext.getProperty(
+                        ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_PASSWORD);
+        String authenticationToken = "Basic " + Base64.getEncoder().encodeToString(
+                authenticationString.getBytes(StandardCharsets.UTF_8));
+        boolean executionStatus = true;
+        String agentLink = tinkererHost + "test-plan/" + testPlan.getId() + "/agents";
```

Comment:
Shall we use Paths (https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html) to derive the URL?

Reply:
fixed